v2/internal/reflecthelpers/reflect_helpers.go

```diff
@@ -282,3 +282,74 @@ func getItemsField(listPtr client.ObjectList) (reflect.Value, error) {
 
 	return itemsField, nil
 }
+
+// SetProperty sets the property on the provided object to the provided value.
+// obj is the object to modify.
+// propertyPath is a dot-separated path to the property to set.
+// value is the value to set the property to.
+// Returns an error if any of the properties in the path do not exist, if the property is not settable,
+// or if the value provided is incompatible.
+func SetProperty(obj any, propertyPath string, value any) error {
```

Comment:
There might now be some duplicate (less generic version) of this code in the `setChildCollection` methods of some of the extensions.

Not clear it's actually fully gonna work with this because it needs to construct items in a collection but figured I'd highlight it anyway.